# run SPAN for QKO samples
```bash
java -Xmx8G -jar span.jar analyze -t $ChIP.bam -c $Control.bam --cs $mm10.chrom.sizes -p $Results.peak
```

# keep only overlapping peaks between replicates
```bash
bedtools intersect -a QKO_rep1.SPAN.H3K36me2_peaks.bed -b QKO_rep2.SPAN.H3K36me2_peaks.bed > QKO_merged_reps_1_2.SPAN.H3K36me2_peaks.bed
bedtools intersect -a QKO_merged_reps_1_2.SPAN.H3K36me2_peaks.bed -b QKO_rep3.SPAN.H3K36me2_peaks.bed > QKO_merged.SPAN.H3K36me2_peaks.bed
```

# Run ChIPseeker
```r
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(org.Mm.eg.db)
library(ChIPseeker)
library(tidyverse)
library(rtracklayer)
library(EnsDb.Mmusculus.v79)
library(EnsDb.Mmusculus.v75)
library(BSgenome.Mmusculus.UCSC.mm10)
library(annotate)
library(gprofiler2)
library(clusterProfiler)
txdb <- TxDb.Mmusculus.UCSC.mm10.knownGene
endb <- EnsDb.Mmusculus.v75
options(ChIPseeker.ignore_downstream = TRUE)
options(ChIPseeker.ignore_5UTR = TRUE)
options(ChIPseeker.ignore_3UTR = TRUE)
options(ChIPseeker.ignore_1st_exon = TRUE)
options(ChIPseeker.ignore_1st_intron = TRUE)

b <- import.bed("10T.QKO.peaks.intersect.merged.bed")
ann <- annotatePeak(b,TxDb = txdb,genomicAnnotationPriority = c("Intergenic","Promoter","Exon","Intron"),tssRegion = c(-1000, 1000))
plotAnnoPie(ann,col = c("indianred1","tan4","navajowhite4","black"))
```